eland/ml/imported_ml_model.py

```diff
@@ -150,7 +155,15 @@ def __init__(
         self._model_type = transformer.model_type
         serializer = transformer.transform()
 
+        if es_if_exists:
+            self.delete_model()
+
         if overwrite:
```

Comment:
We should error out with `ValueError` if both `overwrite` and `es_if_exists` are given.

Reply:
Please help me out. I have completed the logic but facing one issue
**How can we check if both are passed during the object creation?**
Overwrite can take None/True/False
I tried many combinations: Either raising valueerror is failing or the below logic isn't getting executed.
```python
# Since we are  defaulting overwrite  = None and es_if_exists = 'fail'
    if not (isinstance(overwrite,type(None)) and es_if_exists in ['fail','replace']):
        raise ValueError("Use es_if_exists(params 'fail' or 'replace') or overwrite(**Deprecated**).")
# Logic
if not overwrite: 
    es_if_exists = 'replace'

if es_if_exists == 'fail':
   # raise error by checking.
elif es_if_exists == 'replace'
   # delete model
```

Reply:
Can we replace the overwrite with es_if_exists to avoid complexity?

Reply:
This is what I'd do:

```python
def __init__(self, overwrite=None, es_if_exists=None):
    if overwrite is not None and es_if_exists is not None:
        raise ValueError("Using 'overwrite' and 'es_if_exists' together is invalid, use only 'es_if_exists'")
    if overwrite is not None:
        warnings.warn(
            f"'overwrite' parameter is deprecated, use 'es_if_exists' instead",
            DeprecationWarning,
        )
        es_if_exists = "replace" if overwrite else "fail"
    elif es_if_exists is None:
        es_if_exists = "fail"
    if es_if_exists not in ("fail", "replace"):
        raise ValueError("'es_if_exists' must be either 'fail' or 'replace'")
    
```